posts/2023-08-23-liberty-olo-rh-sso.adoc

```diff
@@ -0,0 +1,394 @@
+
+Log in to the KeyCloak from this url.
+
+The credentials for logging in are in the secret - credential-example-keycloak (in the rh-sso nampspace)
+
+Get the secret password from the console for the admin username in this secret
+
+Use this username and password to logn on to the KeyCloak.
+
+After logging in, create a realm, called - "sso-realm".
+This is the url to access this realm
+https://keycloak-rh-sso.apps.<cluster-name>/auth/admin/master/console/#/realms/sso-realm
+
+Next we need to create users (non-admin) for this realm. We can use them to test social login when RH-SSO is used as OIDC provider, the non-admin users can be used to log in to the client application. 
+
+Following the steps to create a user (with it's password)
+testuser1/testpasswd1
+
+Create a user called "testuser1"
+Here are the steps to create the users:
+
+* After logging in Select Manage -> Users and select Add user.
+* Put in the value "testuser1" for Username, and click on Save.
+* After saving, on the next page, select Credentials tab. Enter the password on the next page and ensure to change 'ON' to 'Off' for 'Temporary'. Click on "Reset Password", and on the confirmation dialog, click on "Change Password".
+* Then go to the "Role Mappings" tab. On the Role Mappings page, under Client Roles' drop-down, select realm-management.
+* After 'realm-management' role is selected, additional boxes such as 'Available Roles' appears. Under Available Roles, search for 'view-realm' and select Add selected.
+* After the role is selected, it appears under 'Assigned Roles' and 'Effective Roles'.
+* Note: Selecting the role is just a basic requirement to allow the user to login to the user's console on RH-SSO. If without any role assigned, the user will get Forbidden error msg on the browser after login.
+
+Use the url below to test the users you just created (testuser1)
+
+https://keycloak-rh-sso.apps.<cluster-name>/auth/admin/Sso-realm/console/
+
+You should be able to log in successfully using the testuser1/testpasswd1.
+
+After logging in, in the General Section, you should see the endpoints. 
+
+Click on the link for the OpenID Endpoint Configuration - and that should point you to -
+https://keycloak-rh-sso.apps.<cluster-name>/auth/realms/sso-realm/.well-known/openid-configuration
+
+This will be needed for the client registration as the discoveryEndpoint later on.
+
+== Create the olapp-sso secret 
+
+Create a new project - called `gsm-test`, and create a secret in that namespace - - (Workloads->Secrets->Create Secret), called `guide-social-media-login-olapp-sso`, using key `oidc-clientId` and value `gsmapp`.
+
+The key name should be in this format <app-name>-olapp-sso. Use the same <app-name> as what you will use while deploying the applicaton using OLO.
+
+For e.g. in this setup, the <app-name> from the yaml file is `guide-social-media-login`, so the secret name is `guide-social-media-login-olapp-sso`
```

Comment:
for e.g. doesn't make sense, needs to be changed to "For example, in the application used here in this blog, the <app-name>...."

Reply:
If we're following the example app for this tutorial, can we just assume that it has to be this name and instruct them to use this instead of explaining that that's what it is for this blog? Make it more explicit?